src/src/rx_main.cpp

```diff
@@ -93,62 +104,84 @@ void ICACHE_RAM_ATTR getRFlinkInfo()
     crsf.PackedRCdataOut.ch15 = UINT10_to_CRSF(map(LastRSSI, -100, -50, 0, 1023));
     crsf.PackedRCdataOut.ch14 = UINT10_to_CRSF(fmap(linkQuality, 0, 100, 0, 1023));
 
-    int32_t rssiDBM = LPF_UplinkRSSI.update(Radio.GetLastPacketRSSI());
+    int32_t rssiDBM = LPF_UplinkRSSI.update(Radio.LastPacketRSSI);
     // our rssiDBM is currently in the range -128 to 98, but BF wants a value in the range
     // 0 to 255 that maps to -1 * the negative part of the rssiDBM, so cap at 0.
     if (rssiDBM > 0)
         rssiDBM = 0;
     crsf.LinkStatistics.uplink_RSSI_1 = -1 * rssiDBM; // to match BF
 
     crsf.LinkStatistics.uplink_RSSI_2 = 0;
-    crsf.LinkStatistics.uplink_SNR = Radio.GetLastPacketSNR() * 10;
+    crsf.LinkStatistics.uplink_SNR = Radio.LastPacketSNR * 10;
     crsf.LinkStatistics.uplink_Link_quality = linkQuality;
-    crsf.LinkStatistics.rf_Mode = 4 - ExpressLRS_currAirRate->enum_rate;
+    crsf.LinkStatistics.rf_Mode = 4 - ExpressLRS_currAirRate_Modparams->enum_rate;
 
     //Serial.println(crsf.LinkStatistics.uplink_RSSI_1);
 }
 
 void ICACHE_RAM_ATTR SetRFLinkRate(expresslrs_RFrates_e rate) // Set speed of RF link (hz)
 {
-    expresslrs_mod_settings_s *const mode = get_elrs_airRateConfig(rate);
-    Radio.StopContRX();
-    Radio.Config(mode->bw, mode->sf, mode->cr, Radio.currFreq, Radio._syncWord);
-    ExpressLRS_currAirRate = mode;
-    hwTimer.updateInterval(mode->interval);
-    LPF_PacketInterval.init(mode->interval);
-    //LPF_Offset.init(0);
-    //InitHarwareTimer();
+    expresslrs_mod_settings_s *const ModParams = get_elrs_airRateConfig(rate);
+    expresslrs_rf_pref_params_s *const RFperf = get_elrs_RFperfParams(rate);
+
+    Radio.Config(ModParams->bw, ModParams->sf, ModParams->cr, ModParams->PreambleLen);
+    hwTimer.updateInterval(ModParams->interval);
+
+    ExpressLRS_currAirRate_Modparams = ModParams;
+    ExpressLRS_currAirRate_RFperfParams = RFperf;
+
     Radio.RXnb();
 }
 
+void ICACHE_RAM_ATTR SetTLMRate(expresslrs_tlm_ratio_e TLMrateIn)
+{
+    ExpressLRS_currAirRate_Modparams->TLMinterval = TLMrateIn;
+}
+
 void ICACHE_RAM_ATTR HandleFHSS()
 {
-    uint8_t modresult = (NonceRXlocal + 1) % ExpressLRS_currAirRate->FHSShopInterval;
-    if (modresult != 0)
+    if ((ExpressLRS_currAirRate_Modparams->FHSShopInterval == 0) || alreadyFHSS == true)
     {
         return;
     }
 
-    linkQuality = getRFlinkQuality();
-    if (connectionState != disconnected) // don't hop if we lost
+    uint8_t modresult = (NonceRX + 1) % ExpressLRS_currAirRate_Modparams->FHSShopInterval;
+
+    if ((modresult != 0) || (connectionState == disconnected)) // don't hop if disconnected
+    {
+        return;
+    }
+
+    alreadyFHSS = true;
+    Radio.SetFrequency(FHSSgetNextFreq());
+
+    if (ExpressLRS_currAirRate_Modparams->TLMinterval == TLM_RATIO_NO_TLM)
     {
-        Radio.SetFrequency(FHSSgetNextFreq());
         Radio.RXnb();
+        return;
```

Comment:
Not sure we need these...? Its an `else if`, so nothing else will get executed after this code...